### ISSUE TITLE
The shaded arrow class should not be passed to Spark

### ISSUE THREAD

**Issue by liujiayi771** [oap-project/gluten#1416]
**Describe the bug**
Some code in ArrowWritableColumnVector.java pass a shaded arrow class to Spark method, this will result in a no such method exception.
https://github.com/oap-project/gluten/blob/e6def13f50d86353e7e059b1bb881bf8c5ef66b5/gluten-data/src/main/java/io/glutenproject/vectorized/ArrowWritableColumnVector.java#L1159

```
java.lang.RuntimeException: Error during calling Java code from native code: java.lang.NoSuchMethodError: org.apache.spark.sql.vectorized.ArrowColumnVector.<init>(Lio/glutenproject/shaded/org/apache/arrow/vector/ValueVector;)V
	at io.glutenproject.vectorized.ArrowWritableColumnVector$ArrayAccessor.<init>(ArrowWritableColumnVector.java:1159)
	at io.glutenproject.vectorized.ArrowWritableColumnVector.createVectorAccessor(ArrowWritableColumnVector.java:258)
	at io.glutenproject.vectorized.ArrowWritableColumnVector.<init>(ArrowWritableColumnVector.java:191)
	at io.glutenproject.vectorized.ArrowWritableColumnVector.reserveNewColumn(ArrowWritableColumnVector.java:350)
	at io.glutenproject.vectorized.ArrowWritableColumnVector.reserveNewColumn(ArrowWritableColumnVector.java:77)
	at org.apache.spark.sql.execution.vectorized.WritableColumnVector.<init>(WritableColumnVector.java:838)
	at org.apache.spark.sql.execution.vectorized.WritableColumnVectorShim.<init>(WritableColumnVectorShim.java:39)
	at io.glutenproject.vectorized.ArrowWritableColumnVector.<init>(ArrowWritableColumnVector.java:160)
	at io.glutenproject.vectorized.ArrowWritableColumnVector.<init>(ArrowWritableColumnVector.java:155)
	at io.glutenproject.vectorized.ArrowWritableColumnVector.allocateColumns(ArrowWritableColumnVector.java:110)
	at io.glutenproject.execution.GlutenRowToArrowColumnarExec$$anon$1.javaConvert(GlutenRowToArrowColumnarExec.scala:399)
	at io.glutenproject.execution.GlutenRowToArrowColumnarExec$$anon$1.next(GlutenRowToArrowColumnarExec.scala:423)
	at io.glutenproject.execution.GlutenRowToArrowColumnarExec$$anon$1.next(GlutenRowToArrowColumnarExec.scala:331)
	at io.glutenproject.vectorized.CloseableColumnBatchIterator.next(CloseableColumnBatchIterator.scala:56)
	at io.glutenproject.vectorized.CloseableColumnBatchIterator.next(CloseableColumnBatchIterator.scala:32)
```

**To Reproduce**
we can use a json file to reproduce this problem. Create a json fie test.json
```json
{ "a": { "a1": 1, "a2": "test", "a3": { "a31": "{\"key1\":\"value1\",\"key2\":\"value2\"}" } } }
{"b":{"b1":1,"b2":2}}
```
Use spark-shell to read the json file, and use a function that gluten does not support to generate an r2c, for example input_file_name().
```scala
val df = spark.read.json("test.json")
df.withColumn("input", input_file_name()).show
```
And then you will see the java.lang.NoSuchMethodError.

**Comment by liujiayi771** [comment]
@jinchengchenghh This is the question I told you about.
@zhztheplayer Could you help to resolve this conflict?

**Comment by liujiayi771** [comment]
Can we just exclude this java file?

**Comment by Yohahaha** [comment]
Another case
```
CREATE TABLE default.test (
  id INT,
  date STRING)
USING parquet;

select array(id) from default.test;
```
Error log
```
java.lang.NoSuchMethodError: org.apache.spark.sql.vectorized.ArrowColumnVector.<init>(Lio/glutenproject/shaded/org/apache/arrow/vector/ValueVector;)V
        at io.glutenproject.vectorized.ArrowWritableColumnVector$ArrayAccessor.<init>(ArrowWritableColumnVector.java:1159) ~[gluten-velox-bundle-spark3.3_2.12-alinux_3-0.5.0-SNAPSHOT.jar:?]
        at io.glutenproject.vectorized.ArrowWritableColumnVector.createVectorAccessor(ArrowWritableColumnVector.java:258) ~[gluten-velox-bundle-spark3.3_2.12-alinux_3-0.5.0-SNAPSHOT.jar:?]
        at io.glutenproject.vectorized.ArrowWritableColumnVector.<init>(ArrowWritableColumnVector.java:172) ~[gluten-velox-bundle-spark3.3_2.12-alinux_3-0.5.0-SNAPSHOT.jar:?]
        at io.glutenproject.vectorized.ArrowWritableColumnVector.<init>(ArrowWritableColumnVector.java:155) ~[gluten-velox-bundle-spark3.3_2.12-alinux_3-0.5.0-SNAPSHOT.jar:?]
        at io.glutenproject.vectorized.ArrowWritableColumnVector.loadColumns(ArrowWritableColumnVector.java:138) ~[gluten-velox-bundle-spark3.3_2.12-alinux_3-0.5.0-SNAPSHOT.jar:?]
        at io.glutenproject.utils.GlutenArrowAbiUtil$.toSparkColumnarBatch(GlutenArrowAbiUtil.scala:75) ~[gluten-velox-bundle-spark3.3_2.12-alinux_3-0.5.0-SNAPSHOT.jar:?]
        at io.glutenproject.utils.GlutenArrowAbiUtil$.importToSparkColumnarBatch(GlutenArrowAbiUtil.scala:51) ~[gluten-velox-bundle-spark3.3_2.12-alinux_3-0.5.0-SNAPSHOT.jar:?]
        at io.glutenproject.utils.GlutenArrowAbiUtil.importToSparkColumnarBatch(GlutenArrowAbiUtil.scala) ~[gluten-velox-bundle-spark3.3_2.12-alinux_3-0.5.0-SNAPSHOT.jar:?]
```

cc @FelixYBW @zhouyuan

**Comment by FelixYBW** [comment]
@zhztheplayer Can you fix?

**Comment by zhztheplayer** [comment]
Will take a look asap.

**Comment by FelixYBW** [comment]
> Will take a look asap.

Any possible we reuse Spark's old arrow? Since the C_Data API doesn't change.

**Comment by jinchengchenghh** [comment]
We still use netty to convert Arrow data to bytes, and columnar to row, maybe we cannot remove it now.
And java arrow version should unify with C version, otherwise we may get some unexpected error.

**Comment by Yohahaha** [comment]
> We still use netty to convert Arrow data to bytes, and columnar to row, maybe we cannot remove it now.

What does Netty do here? I see Netty only be used to read/write bytes between on-heap and off-heap, does Arrow has no ability on this?

>  java arrow version should unify with C version, otherwise we may get some unexpected error.

This is right in most case. But Arrow is cross-language, if IPC interface not changed, we could use different version of cArrow and jArrow in theory, just like bingwei said.

**Comment by zhztheplayer** [comment]
> Any possible we reuse Spark's old arrow? Since the C_Data API doesn't change. C++ use latest Arrow, java use Spark's.

It's possible but more complicated than the way we think of it:

* Java C ABI was added to Arrow since 6.0.0, while Spark 3.2 only depends on 2.0.0
* We have a couple of bug fixes to get Java C ABI work for Gluten, some of them were merged to Apache Arrow in some version later than 6.0.0, even 12.0.0. So we are still using [patched Arrow 11.0.0](https://github.com/oap-project/arrow/tree/arrow-11.0.0-gluten).

So I would expect we can tweek a little bit on the Jar shading part to get this issue fixed.
